examples/experimental/koen/CNN plan.ipynb

```diff
@@ -0,0 +1,1058 @@
+{
```

Comment:
<p>train_pointer is accepting actual variables as input instead of pointers to variables already on the remote machine. This makes me think that the plan is actually being executed locally on pointers instead of the actions actually being sent to the remote machien to be executed there. This would mean that there's still a tremendous amount of network traffic (one message per action) when running a plan as opposed to only saying "hey run this plan on these variables" being the only message actually sent over the network.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5455/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Reply:
<p>Yes, the plan pointer does accept variables (instead of pointers), however I think it would also accept pointers. But it does not mean that those variables are not sent and executed remotely (I am not sure how to interpret "local", is a VM also local?). When we execute the plan pointer by passing variables we don't need a lot of traffic, because we only serialize the arguments, not the actions in the plan. <strong>However, </strong>we currently use a hack for local execution of plans, which actually involves sending them to an ephemeral machine. We should get rid of that hack at some point.</p><p> </p><p>Also, this notebook is outdated </p>